Доработки по комментариям для защиты

KKKonstantin opened the pull request:
1.
Задача:
https://skr.sh/sEvDGjdUabI
В мозилле селекты другого цвета.
Решение:
прописал состояния для селектов - http://joxi.ru/4AkGnQMHo65RoA

2.
Задача:
https://skr.sh/sEvgWWoO3UI

Решение:
переполненность навигатора решил через margin-bottom, flex-wrap: wrap и у блока есть жесткая ширина, чтобы текст спускался вниз, получается так - http://joxi.ru/a2XO9DLfwb7Vym

3.
Задача:
https://skr.sh/sEvxLYjM0Hb
квадраты должны тянуться под цифры
Решение: прописал для ячеек display: inline-table, и выглядит теперь так - http://joxi.ru/ZrJp94buwpN6e2, я не знаю что это значит, но работает, через max-width не удалось решить(

4.
Задача: https://skr.sh/sEvlQKRlFz0 - Нормализация тега img отсутствует, картинки бОльшего размера будут выпадать
Решение: прописал такие параметры - http://joxi.ru/nAy9GQOhg1W3wA

5.
Задача:
https://skr.sh/sEv467hEo99
https://skr.sh/sEv6Uxbg3fO
https://skr.sh/sEvXRKPgiwe

И где модалка?..

Решение: с использованием Pixel Perfect привел страницы сайта в соответствие макету:
http://joxi.ru/KAgWa0NsEvV97A
http://joxi.ru/8AnBkWQFzl8nnA
http://joxi.ru/KAxjlX3IZk6Rpm

6.
Задача: https://skr.sh/sEv58UxoFOg
Фокус не по макету + паддинг слева надо добавить чтобы печатал пользователь не с самого начала, т.к. текст прилипать будет к краю, некрасиво.

Решение: прописал состояния согласно этой части стайлгайда - http://joxi.ru/L21RElNSR4WLZA, паддинг дополнен, курсор там же где и placeholder - http://joxi.ru/p27oB10FKBD5g2

7.
Задача:
https://skr.sh/sEvixnPySmp
Состояний нет/не по макету
Решение: дополнены все состояния по стайлгайду для этих элементов

8.
Задача:
https://skr.sh/sEvbQ6AQWjN
Неверный ховер
Решение: ховер скорректирован согласно стайлгайду и иные состояния этого элемента

9.
Задача: https://skr.sh/sEvE11fWHuE
Клик по этим кнопкам приводит к отправке формы, что является непредсказуемым поведением и нежелательным тоже. Так происходит потому, что вы не указали type="button" кнопкам.
Решение: по указаниям обоим кнопкам дополнен type="button"

10.
Задача:
https://skr.sh/sEv7AGLqcHn
Не думаю что это лучшее решение. А если карточек 15 на станице будет и все пятизвездочные? 75 картинок грузить будете?
Тут можно свг инлайновым в разметку вставлять и добавлять какой нибудь скрытый спан для доступности, в котором будет текст типа 5 звезд или 4 звезды. Скорость загрузки будет выше у страницы и способ удобнее.
Решение: добавлены svg-картинки, выравнены по макету, а также span с описанием

11.
Задача:
https://skr.sh/sEvf805jyo6

Решение: дополнены скрытые описания для некоторых svg, для <a> не присваивал - валидатор ругается.

12.
Задача:
https://skr.sh/sEvb3nQAoYV
Не хватает фокуса чекбоксам и радиокнопкам.
Решение: чекбоксам и радиокнопкам присвоены все состояния согласно стайлгайду.

KKKonstantin committed:
Доработки по комментариям для защиты

1.
Задача:
https://skr.sh/sEvDGjdUabI
В мозилле селекты другого цвета.
Решение:
прописал состояния для селектов - http://joxi.ru/4AkGnQMHo65RoA

2.
Задача:
https://skr.sh/sEvgWWoO3UI

Решение:
переполненность навигатора решил через margin-bottom, flex-wrap: wrap и у блока есть жесткая ширина, чтобы текст спускался вниз, получается так - http://joxi.ru/a2XO9DLfwb7Vym

3.
Задача:
https://skr.sh/sEvxLYjM0Hb
квадраты должны тянуться под цифры
Решение: прописал для ячеек display: inline-table, и выглядит теперь так - http://joxi.ru/ZrJp94buwpN6e2, я не знаю что это значит, но работает, через max-width не удалось решить(

4.
Задача: https://skr.sh/sEvlQKRlFz0 - Нормализация тега img отсутствует, картинки бОльшего размера будут выпадать
Решение: прописал такие параметры - http://joxi.ru/nAy9GQOhg1W3wA

5.
Задача:
https://skr.sh/sEv467hEo99
https://skr.sh/sEv6Uxbg3fO
https://skr.sh/sEvXRKPgiwe

И где модалка?..

Решение: с использованием Pixel Perfect привел страницы сайта в соответствие макету:
http://joxi.ru/KAgWa0NsEvV97A
http://joxi.ru/8AnBkWQFzl8nnA
http://joxi.ru/KAxjlX3IZk6Rpm

6.
Задача: https://skr.sh/sEv58UxoFOg
Фокус не по макету + паддинг слева надо добавить чтобы печатал пользователь не с самого начала, т.к. текст прилипать будет к краю, некрасиво.

Решение: прописал состояния согласно этой части стайлгайда - http://joxi.ru/L21RElNSR4WLZA, паддинг дополнен, курсор там же где и placeholder - http://joxi.ru/p27oB10FKBD5g2

7.
Задача:
https://skr.sh/sEvixnPySmp
Состояний нет/не по макету
Решение: дополнены все состояния по стайлгайду для этих элементов

8.
Задача:
https://skr.sh/sEvbQ6AQWjN
Неверный ховер
Решение: ховер скорректирован согласно стайлгайду и иные состояния этого элемента

9.
Задача: https://skr.sh/sEvE11fWHuE
Клик по этим кнопкам приводит к отправке формы, что является непредсказуемым поведением и нежелательным тоже. Так происходит потому, что вы не указали type="button" кнопкам.
Решение: по указаниям обоим кнопкам дополнен type="button"

10.
Задача:
https://skr.sh/sEv7AGLqcHn
Не думаю что это лучшее решение. А если карточек 15 на станице будет и все пятизвездочные? 75 картинок грузить будете?
Тут можно свг инлайновым в разметку вставлять и добавлять какой нибудь скрытый спан для доступности, в котором будет текст типа 5 звезд или 4 звезды. Скорость загрузки будет выше у страницы и способ удобнее.
Решение: добавлены svg-картинки, выравнены по макету, а также span с описанием

11.
Задача:
https://skr.sh/sEvf805jyo6

Решение: дополнены скрытые описания для некоторых svg, для <a> не присваивал - валидатор ругается.

12.
Задача:
https://skr.sh/sEvb3nQAoYV
Не хватает фокуса чекбоксам и радиокнопкам.
Решение: чекбоксам и радиокнопкам присвоены все состояния согласно стайлгайду.

pavel-nyukalo commented on catalog.html:
```diff
@@ -168,11 +169,14 @@ <h2 class="visually-hidden">Результаты поиска.</h2>
-              <option value="cheap">Сначала дешёвое</option>
-              <option value="expensive">Сначала дорогое</option>
-            </select>
+            <div class="select-control-wrapper">
```
[TEST-01](https://up.htmlacademy.ru/htmlcss/34/criteries#TEST-01)
Проверяющий скорее всего имел ввиду, что в Firefox селект имеет серый фоновый цвет.

Сейчас так же:
![image](https://user-images.githubusercontent.com/52506235/179358663-b4371bac-6fed-4598-a899-1069b4ea2bd5.png)

Стоит явно прописать ему `background-color: white (в том формате, в котором пишешь по всему проекту)`

pavel-niukalo reviewed (changes requested):
[TEST-04](https://up.htmlacademy.ru/htmlcss/34/criteries#TEST-04)
Стоит добавить либо отступ, либо минимальную высоту, чтобы черта не залезала на другой элемент  
![image](https://user-images.githubusercontent.com/52506235/179358796-e9fbe41f-4f60-44d9-be84-f8b214752620.png)

[TEST-06](https://up.htmlacademy.ru/htmlcss/34/criteries#TEST-06)

- Белую маску эту тоже надо добавить (через множественный фон)

![image](https://user-images.githubusercontent.com/52506235/179358921-dceb6062-5695-4923-a1de-1f14dbfeee5b.png)

- На макете черный цвет

![image](https://user-images.githubusercontent.com/52506235/179358956-5a28a160-4de1-4b1a-817f-bf0d5e6f48f5.png)

- На это тоже могут обратить внимание

![image](https://user-images.githubusercontent.com/52506235/179358991-7caeb412-0e4f-4529-a910-6513d055d734.png)

- На макете там другая фоновая картинка

Сейчас:

![image](https://user-images.githubusercontent.com/52506235/179359012-27bc5d8b-bc10-4df8-a680-561199536ed9.png)

Макет:

![image](https://user-images.githubusercontent.com/52506235/179359021-da8ac78e-5cff-420a-bfd4-62e37e101c3c.png)

- Надо добавить модалку

[TEST-07](https://up.htmlacademy.ru/htmlcss/34/criteries#TEST-07)

- Стоит убрать браузерную обводку

![image](https://user-images.githubusercontent.com/52506235/179359168-933dc697-20df-451a-9cb9-7ec5297783a0.png)

- Стоит добавить паддинг не только слева, но и справа, чтобы текст не упирался в кнопку "Подписаться"

- Сейчас при фокусе и нажатии добавляется `border`. `border` тоже участвует в блочной модели, поэтому текст немного прыгает.

Чтобы это решить можно использовать:
а. Задать `border` сразу и сделать его прозрачным, а при фокусе и нажатии менять только цвет `border-color`
б. Использовать `box-shadow`, можно внутренний, можно внешний
в. Использовать `outline`, но там проблемы могут быть с скруглением

Наиболее простой это первый вариант с `border`

- Неверное фокусное состояние

![image](https://user-images.githubusercontent.com/52506235/179359326-573129c8-1342-43cb-acde-a964eee25aef.png)

Надо сбросить браузерное и сделать при помощи `box-shadow`, у меня есть [заготовка](https://codesandbox.io/s/input-99kzwz) 

- Вот эта кнопка в фокусе тоже начинает прыгать

![image](https://user-images.githubusercontent.com/52506235/179359377-835f35b3-df29-4131-b681-c7fb6a19aaf1.png)

Тут та же проблема с `border`

[ALLY-02](https://up.htmlacademy.ru/htmlcss/34/criteries#ALLY-02)

- В каталоге забыл добавить подпись
![image](https://user-images.githubusercontent.com/52506235/179359524-bb62603f-cb59-44e3-9357-2c9615317aa1.png)

[ALLY-03](https://up.htmlacademy.ru/htmlcss/34/criteries#ALLY-03)

- У чекбоксов так же не хватает фокуса при использовании Tab

По-хорошему кружок надо сделать спаном и спозиционировать, потому что цифра может меняться
![image](https://user-images.githubusercontent.com/52506235/179359673-545451ab-19f3-4f04-8ead-a83250e31eed.png)

KKKonstantin committed:
Повторно дорабатывает код перед защитой

1. добавил отступ снизу, теперь элементы не мешают друг другу при переполнении - http://joxi.ru/vAWy9MdcgWyLYm
2. поправил расположение кнопки Сбросить
3. поправил цвет выбранной цифры в пагинации
4. поменял бэкграунд у каталога
5. модалку сделал - прошу проверить, там могут быть отклонения по пискселям - давай плс отметим наиболее критичные моменты и я их с PP пофикшу
6. в ячейке с кнопкой подписаться добавил паддинг справа, поправил бордеры у ячейки с кнопкой подписаться
7. скорректировал фокусное состояние ячейки с ценой по комментам
8. скорректировал бордер у кнопки Сбросить
9. добавил и в каталоге подпись для лого академии
10. у чекбоксов есть фокус в коде и в инспекторе он работает, при этом если жмем Таб, то он не работает - не смог разобраться в чем дело - помоги плс если получится
11. сделал кружок над сердечком в виде спана и указан значение 12, которое может изменяться в html
12. добавил в селект белый цвет фона

+ внес несколько других правок, посмотрите плс

KKKonstantin committed:
Дорабатывает код по критериям

Дополнительно внесены изменения согласно TEST-03. Размеры страницы., HTML-03. Кнопки и ссылки.,  HTML-04. Работоспособные формы., HTML-05. Подписи полей форм., ALLY-02. Подписи интерактивных элементов.

pavel-nyukalo commented on catalog.html:
```diff
@@ -67,30 +69,31 @@ <h1 class="catalog-filter-title">Гостиницы Седоны</h1>
-                          disabled>
+                      <label class="control" for="option-pool">
+                        <input class="visually-hidden control-input" type="checkbox" name="option-pool" id="option-pool"
+                          checked disabled>
```
`disabled` не забудь убрать перед отправкой на защиту

pavel-nyukalo commented on index.html:
```diff
@@ -196,10 +204,168 @@ <h2 class="subscribe-title">Подпишитесь на рассылку</h2>
+                <span class="visually-hidden">Календарь.</span>
+
+              </p>
+              <p class="field-group field-text">Дата Выезда:</p>
```
Аналогично

pavel-nyukalo commented on index.html:
```diff
@@ -196,10 +204,168 @@ <h2 class="subscribe-title">Подпишитесь на рассылку</h2>
+              </p>
+
+              <div class="field-command field-command-adult">
+                <p class="field-command field-text">Взрослые:</p>
```
Это тоже `label`

KKKonstantin committed:
Дорабатывает код перед защитой

KKKonstantin committed:
Дорабатывает по комментариям для защиты проекта

KKKonstantin committed:
Дорабатывает страницы сайта по комментариям, а также дополняет код в результате добавления автопрефиксов

pavel-nyukalo commented:
![image](https://user-images.githubusercontent.com/52506235/181310408-eb09af1d-95d3-4d39-9df2-90e8689e00c7.png)
